modules/social_features/social_private_message/social_private_message.module

```diff
@@ -618,3 +636,26 @@ function social_private_message_views_query_alter(ViewExecutable $view, QueryPlu
     $query->addWhere(NULL, 'pm_thread_delete_time.owner', $current_user_id, '=');
   }
 }
+
+/**
+ * Implements hook_private_message_thread_access().
+ */
+function social_private_message_private_message_thread_access(PrivateMessageThreadInterface $entity, $operation, AccountInterface $account) {
+  if ($operation === 'delete') {
+    if (
+      $account->hasPermission('use private messaging system') &&
+      $account->hasPermission('delete private messages thread')
+    ) {
+      return AccessResult::allowed();
+    }
+
+    return AccessResult::forbidden();
+  }
+}
+
+/**
+ * Implements hook_private_message_access().
```

Comment:
@dmytriikaiun , could you change "Implements hook_private_message_access()" to "Implements hook_private_message_create_access()", please